### ISSUE TITLE
Adding a -to-many relationship then removing it blocks merging

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3872]
https://github.com/specify/specify7/assets/37256050/c4c37076-a9fb-4426-82d0-9de08b634c02

`herb_rbge` database

http://localhost/specify/overlay/merge/Agent/?records=47121%2C47120%2C47119%2C47118%2C47117%2C47116%2C47115%2C47114%2C47113%2C47112%2C47111%2C47110%2C47109%2C47108%2C47107%2C47106%2C47105

**Steps to Recreate:**
1. Uncheck "Show conflicting fields only"
1. Go to the **Agent Attachments** button
2. Add a blank record
3. Remove it
4. See merge is now blocked by the red button

**Comment by melton-jason** [comment]
This issue occurred whenever a saveblocker was added to the dependent resource (in this case, `ordinal` and `agent` are required for an AgentAttachment). 
As soon as the AgentAttachment resource is created, it adds the saveblockers for all of the required fields. 
To handle -to-many resources, we create a list (array) based off of the resources in Backbone's internal array for that collection. 

The trash icon just removed the resource from the array we created. 
We have to tell Backbone that we are deleting the Resource we created as well, which will remove the saveBlockers.
